File: DevExpress-Demo/CS/ChartsDemo.Wpf/DataModels/OperatingSurfaceTemperatureData.cs
using System;
using System.Collections.Generic;

namespace ChartsDemo {

    public class OperatingSurfaceTemperatureData {
        const int PointsCount = 250;
        readonly TemperaturePoint maxTemperaturePoint = new TemperaturePoint(TimeSpan.Zero, double.MinValue);
        readonly TemperaturePoint minTemperaturePoint = new TemperaturePoint(TimeSpan.MaxValue, double.MaxValue);
        readonly List<TemperaturePoint> temperaturePoints = new List<TemperaturePoint>(PointsCount);

        public double MinOptimalTemperature { get { return 43; } }
        public double MaxOptimalTemperature { get { return 63; } }
        public double OptimalTemperature { get { return 53; } }
        public TemperaturePoint MaxTemperaturePoint {
            get { return this.maxTemperaturePoint; }
        }
        public TemperaturePoint MinTemperaturePoint {
            get { return this.minTemperaturePoint; }
        }
        public List<TemperaturePoint> TemperaturePoints {
            get { return this.temperaturePoints; }
        }

        public OperatingSurfaceTemperatureData() {
            Random random = new Random(9);
            double preTemperature = 50;
            for (int i = 0; i < PointsCount; i++) {
                TimeSpan time = TimeSpan.FromSeconds(i);
                double temperature = preTemperature + (random.NextDouble() - 0.5) * 10;
                if (temperature > 90)
                    temperature -= 20;
                if (temperature < 20)
                    temperature += 10;
                TemperaturePoint temperaturePoint = new TemperaturePoint(time, temperature);
                if (temperature < this.minTemperaturePoint.Temperature)
                    this.minTemperaturePoint = temperaturePoint;
                if (temperature > this.maxTemperaturePoint.Temperature)
                    this.maxTemperaturePoint = temperaturePoint;
                this.temperaturePoints.Add(temperaturePoint);
                preTemperature = temperature;
            }
        }
    }


    public class TemperaturePoint {
        public TimeSpan TimeStamp { get; private set; }
        public double Temperature { get; private set; }

        internal TemperaturePoint(TimeSpan time, double temperature) {
            TimeStamp = time;
            Temperature = temperature;
        }
    }

}

File: DevExpress-Demo/VB/ChartsDemo.Wpf/DataModels/OperatingSurfaceTemperatureData.vb
Imports System
Imports System.Collections.Generic

Namespace ChartsDemo

    Public Class OperatingSurfaceTemperatureData

        Const PointsCount As Integer = 250

        Private ReadOnly maxTemperaturePointField As TemperaturePoint = New TemperaturePoint(TimeSpan.Zero, Double.MinValue)

        Private ReadOnly minTemperaturePointField As TemperaturePoint = New TemperaturePoint(TimeSpan.MaxValue, Double.MaxValue)

        Private ReadOnly temperaturePointsField As List(Of TemperaturePoint) = New List(Of TemperaturePoint)(PointsCount)

        Public ReadOnly Property MinOptimalTemperature As Double
            Get
                Return 43
            End Get
        End Property

        Public ReadOnly Property MaxOptimalTemperature As Double
            Get
                Return 63
            End Get
        End Property

        Public ReadOnly Property OptimalTemperature As Double
            Get
                Return 53
            End Get
        End Property

        Public ReadOnly Property MaxTemperaturePoint As TemperaturePoint
            Get
                Return maxTemperaturePointField
            End Get
        End Property

        Public ReadOnly Property MinTemperaturePoint As TemperaturePoint
            Get
                Return minTemperaturePointField
            End Get
        End Property

        Public ReadOnly Property TemperaturePoints As List(Of TemperaturePoint)
            Get
                Return temperaturePointsField
            End Get
        End Property

        Public Sub New()
            Dim random As Random = New Random(9)
            Dim preTemperature As Double = 50
            For i As Integer = 0 To PointsCount - 1
                Dim time As TimeSpan = TimeSpan.FromSeconds(i)
                Dim temperature As Double = preTemperature + (random.NextDouble() - 0.5) * 10
                If temperature > 90 Then temperature -= 20
                If temperature < 20 Then temperature += 10
                Dim temperaturePoint As TemperaturePoint = New TemperaturePoint(time, temperature)
                If temperature < minTemperaturePointField.Temperature Then minTemperaturePointField = temperaturePoint
                If temperature > maxTemperaturePointField.Temperature Then maxTemperaturePointField = temperaturePoint
                temperaturePointsField.Add(temperaturePoint)
                preTemperature = temperature
            Next
        End Sub
    End Class

    Public Class TemperaturePoint

        Private _TimeStamp As TimeSpan, _Temperature As Double

        Public Property TimeStamp As TimeSpan
            Get
                Return _TimeStamp
            End Get

            Private Set(ByVal value As TimeSpan)
                _TimeStamp = value
            End Set
        End Property

        Public Property Temperature As Double
            Get
                Return _Temperature
            End Get

            Private Set(ByVal value As Double)
                _Temperature = value
            End Set
        End Property

        Friend Sub New(ByVal time As TimeSpan, ByVal temperature As Double)
            TimeStamp = time
            Me.Temperature = temperature
        End Sub
    End Class
End Namespace
